src/client/Client.cc

```diff
@@ -15975,6 +15978,14 @@ void Client::handle_conf_change(const ConfigProxy& conf,
     _collect_and_send_global_metrics = cct->_conf.get_val<bool>(
       "client_collect_and_send_global_metrics");
   }
+  if (changed.count("client_caps_release_delay")) {
+    caps_release_delay = cct->_conf.get_val<std::chrono::seconds>(
+      "client_caps_release_delay");
+  }
+  if (changed.count("client_mount_timeout")) {
+    mount_timeout = cct->_conf.get_val<std::chrono::seconds>(
+      "client_mount_timeout");
```

Comment:
I think the root cause is here.

The `mount_timeout` usually will be `0` unless you have changed it during the `ceph daemon client.XXX config set client_mount_timeout 300` command during the daemon is running.

You should also assign it in the `Client::mount()` IMO.